ansible_builder/requirements.py

```diff
@@ -14,24 +14,28 @@
 ))
 
 
-def sanitize_requirements(py_reqs):
-    parsed = requirements.parse('\n'.join(py_reqs))
-
+def sanitize_requirements(collection_py_reqs):
     # de-duplication
     consolidated = []
     seen_pkgs = set()
-    for req in parsed:
-        if req.name is None:
-            consolidated.append(req)
-            continue
-        if req.name in seen_pkgs:
-            for prior_req in consolidated:
-                if req.name == prior_req.name:
-                    prior_req.specs.extend(req.specs)
-                    break
-            continue
-        consolidated.append(req)
-        seen_pkgs.add(req.name)
+    for collection, lines in collection_py_reqs.items():
+        try:
+            for req in requirements.parse('\n'.join(lines)):
+                req.collections = [collection]  # add backref for later
+                if req.name is None:
+                    consolidated.append(req)
+                    continue
+                if req.name in seen_pkgs:
+                    for prior_req in consolidated:
+                        if req.name == prior_req.name:
+                            prior_req.specs.extend(req.specs)
+                            prior_req.collections.append(collection)
+                            break
+                    continue
+                consolidated.append(req)
+                seen_pkgs.add(req.name)
+        except Exception as e:
+            print('Warning: failed to parse requirments from {}, error: {}'.format(collection, e))
```

Comment:
Added try-except here so that we would always get a result as opposed to erring.

I don't think a single bad-actor collection should halt the entire process. If a collection has stuff that cannot be parsed, we don't want to include any of its requirements, because that's a red flag.